Document FXIOS-11533 [Readmes] Simplify referencing Xcode versions and update links (#25057)
mozilla-mobile/firefox-ios: CONTRIBUTING.md

### Reference Person
Each `Contributor OK` issue typically has a reference person assigned. If you need help or clarification:

> - Reach out on [Mozilla Matrix chat](https://wiki.mozilla.org/Matrix).

Author: Actually there is the "other" link, somewhere down the page with real anchor to the specific room.

(It was the reason for me to avoid the duplication, and provide this wiki page with platform+connection details for complete newcomers as an introduction, while the contact section has the people/channels.)

Totally fine with having both here, only would use the room deeplink instead of the space/home.

> - Reach out on [Mozilla Matrix chat](https://wiki.mozilla.org/Matrix).

Reviewer: Ahh, I see the second reference at the bottom of the page. How about something like this 

```suggestion
- Reach out on the [fx-ios](https://chat.mozilla.org/#/room/#fx-ios:mozilla.org) Matrix channel.
```

and then we add

`For more information on how to get started with Matrix, see the [Mozilla Matrix wiki page](https://wiki.mozilla.org/Matrix)` as a nested bullet under `Chat: See Element channel [#fx-ios](https://chat.mozilla.org/#/room/#fx-ios:mozilla.org) for general discussion. You can also write DMs to specific teammates on it.` at the bottom of the page

We'll have two links to the Matrix channel, but I think that's okay.

> - Reach out on [Mozilla Matrix chat](https://wiki.mozilla.org/Matrix).

Author: (Perhaps de-dupe and just reference the section below:)

```suggestion
- Reach out on [Mozilla Matrix chat](#reaching-out-for-help-and-questions).
```

- `Advanced`: These tasks demand a deeper understanding of the project. They often include complex implementations, significant refactoring, or intricate bug fixes.

## Pull Requests
* All pull requests must be associated with a specific issue. If an issue doesn't exist, please create it first.
* Before you submit your pull request, search the repository for open or closed PRs that relate to your submission. We don't want to duplicate effort. 
* PRs should be made from a branch on your personal fork to the `mozilla-mobile:main` branch. Please see the [Pull Request Naming Guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide) for how to name PRs.

> * For hints how to split and describe your commits, see the ["Commits" section](#commits) below for detailed info.

Reviewer: I am referring to the last bullet point.

> * For hints how to split and describe your commits, see the ["Commits" section](#commits) below for detailed info.

Author: Yes, I was thinking the same myself. Kept it to follow the original pattern, in case it was somehow a "PR requirement", but more than happy to omit.

> * For hints how to split and describe your commits, see the ["Commits" section](#commits) below for detailed info.

Author: ```suggestion
```

If you submit a patch, be sure to exclude these files because they are only relevant for your personal build.

## Contributor Fix
We add the "Contributor Fix" label on tasks that have a PR opened for it, or if a PR has been merged to fix this task. This means if you see this label on a task it's probably fixed and cannot be picked up. Note that tasks still stay opened before we close them as it's the Quality Assurance people that will close those tasks with their final approval of the work.

# Reaching out for help and questions
If more information is required or contributors have any questions then we suggestion reaching out to us via:

> - Chat: See Matrix channel [#fx-ios](https://chat.mozilla.org/#/room/#fx-ios:mozilla.org) for general discussion. You can also write DMs to specific teammates on it.

Author: ```suggestion
- Chat: See Matrix channel [#fx-ios](https://chat.mozilla.org/#/room/#fx-ios:mozilla.org) for general discussion. You can also write DMs to specific teammates on it. (For more information on how to get started with Matrix, see [Mozilla Matrix wiki page](https://wiki.mozilla.org/Matrix).)
```
